docker/pypi/wmagent-mariadb/mariadb-docker-build.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+### This script is to be used for building a MariaDB docker imagge based on pypi
+### It depends on a single parameter MDB_TAG
+
+
+help(){
+    echo -e $*
+    cat <<EOF
+
+The MariaDB docker build script for Docker image creation based on pypi:
+
+Usage: mariadb-docker-build.sh -v <mariadb_tag>
+
+    -t <mariadb_tag>                The MariaDB version/tag to be used for the Docker image creation
+    -p <push_image>                 Push the image to registry.cern.ch
+    -l <latest_tag_toregistry>      Push the curernt tag also as latest to registry.cern.ch
+
+Example: ./mariadb-docker-build.sh -v 2.2.0.2
+
+EOF
+}
+
+usage(){
+    help $*
+    exit 1
+}
+
+MDB_TAG=None
+PUSH=false
+LATEST=false
+
+### Argument parsing:
+while getopts ":t:hpl" opt; do
+    case ${opt} in
+        t) MDB_TAG=$OPTARG ;;
+        p) PUSH=true ;;
+        l) LATEST=true ;;
+        h) help; exit $? ;;
+        \? )
+            msg="Invalid Option: -$OPTARG"
+            usage "$msg" ;;
+        : )
+            msg="Invalid Option: -$OPTARG requires an argument"
+            usage "$msg" ;;
+    esac
+done
+
+
+# NOTE: NO MDB_TAG validation is done in the current script. It is implemented at the install.sh
+
+dockerOpts=" --network=host --progress=plain --build-arg MDB_TAG=$MDB_TAG "
+
+docker build $dockerOpts -t local/mariadb:$MDB_TAG -t local/mariadb:latest  .
+
+$PUSH && {
+    docker login registry.cern.ch
```

Comment:
Hi @vkuznet  I finally made  it work properly.  I added this  option in purpose. Because we are about to get into a mess with local and remote images if we do not maintain a hard tagging schema. So whenever  one  adds the `-p` option to either `mariadb-docker-run.sh` or `mariadb-docker-build.sh` it is always going to be an image pulled/pushed from/to registry.cern.ch to be run or uploaded respectively. let me put it in an example to visualize what  I mean:

Imagine one is working on building a new tag and he/she has uploaded already a version in the registry, but still continued working on the local build and ends up with  two versions of the same tag one uploaded and one local. If he was about to use the wrapper `mariadb-docker-build.sh` he/she would have ended with the following set of images at the host:

```
cmst1@vocms0290:wmagent-mariadb $ docker image  ls 
REPOSITORY                 TAG       IMAGE ID       CREATED          SIZE
local/mariadb              10.6.5    4efa646aea3e   6 minutes ago    950MB
local/mariadb              latest    4efa646aea3e   6 minutes ago    950MB
registry.cern.ch/mariadb   10.6.5    8539e03b7a1d   21 minutes ago   950MB
registry.cern.ch/mariadb   latest    8539e03b7a1d   21 minutes ago   950MB
```    

Looking at the repository names it becomes immediately clear which image comes from where. And looking at the image hashes you can clearly see that  the version uploaded at the registry differs from the version at  the local repo.

Now  trying to run those again with the `mariadb-docker-run.sh` wrapper, immediately tells you which is the image you are  currently running and you point to the one you want again just by adding/removing the -p parameter to the command:

Running the local one:
```
cmst1@vocms0290:wmagent-mariadb $ ./mariadb-docker-run.sh -t 10.6.5
Starting the mariadb:10.6.5 docker container with the following parameters:  --user cmst1
eb7e0d879d4d7fa597587c734837c5289886a6aaf6a82c072187371fdf312b90

cmst1@vocms0290:wmagent-mariadb $ docker ps 
CONTAINER ID   IMAGE                  COMMAND      CREATED         STATUS         PORTS     NAMES
eb7e0d879d4d   local/mariadb:10.6.5   "./run.sh"   3 seconds ago   Up 2 seconds             mariadb
```

Running the one from the remote registry:
```
cmst1@vocms0290:wmagent-mariadb $ ./mariadb-docker-run.sh -t 10.6.5 -p
Pulling Docker image: registry.cern.ch/cmsweb/mariadb:10.6.5
10.6.5: Pulling from cmsweb/mariadb
Digest: sha256:61f798b55a1c743686e1568509975308dc07b5b24486894053d6a312983c4af6
Status: Downloaded newer image for registry.cern.ch/cmsweb/mariadb:10.6.5
registry.cern.ch/cmsweb/mariadb:10.6.5
Starting the mariadb:10.6.5 docker container with the following parameters:  --user cmst1
21d9c6598f35e627834d1b796460047605d6255cebc746d572289c7b418053ed

cmst1@vocms0290:wmagent-mariadb $ docker ps 
CONTAINER ID   IMAGE                             COMMAND      CREATED         STATUS         PORTS     NAMES
21d9c6598f35   registry.cern.ch/mariadb:10.6.5   "./run.sh"   7 seconds ago   Up 6 seconds             mariadb
``` 

This is  an approach I have  adopted and gradually polished with all the types of containers we are about to run for the WMAgent.

And I, Yes, I agree with you this may create a mesup with the credentials if many people  use the same user for uploading. That's why with my latest commit:
* I Added an explicit check for the login name of the user to build/connect to the ccern registry and if it is not the current one (i.e. someone have switched to `cmst1` or other user before uploading) the script just aborts and does not proceed with creating a login session. 
* I explicitly destroy the login session at the end of the execution - this will remove the credentials preserved in the user's home.


It is another story if we are willing to configure credential helpers for these operations. I'd say we should, but they do not work properly on our cc7 machines(i.e. the `pass` command is missing in our agents). So as long as we are about to be working with WMAgent images on our current agents, we are bound to temporarily storing credentials in the config file for the time the lgin session is active.  But this should be only during this transient period.  Once we move to alma9 agents  this would be easily fixable as well.